docs/DeveloperGuide.md

```diff
@@ -2,37 +2,615 @@
+     `execute()` method.
+   - If the `keyWord` is `importance` it creates and instance of its subclass `FilterByImportanceCommand` and invoke the
+     `execute()` method.
+- The respective subclasses will print the `CS2040CFiles` filtered based on the `keyWord` provided.
+
+### TopoSort
+**API** : `TopoCommand.java`
+
+Here is a class diagram of the `TopoCommand` which facilitates the storage
+function of the application.
+
+![](.\\uml\\diagrams\\TopoCommandClass.png "TopoCommand Class Diagram")
+
+The `TopoCommand` component
+- can topologically sort `CS2040CFile` objects in a specific `topic` order
+- can print out the list of topologically sorted `CS2040CFile` objects
+- can check whether there are `CS2040CFile` objects within `CLIAlgo` and inform user if no such objects are saved
+
+## Implementation
+
+### Parser
+#### Current Implementation
+Parsing of commands is done by the `Parser` class. It implements the `StringManipulation` interface which allows
+`Parser` to extract the relevant keywords to prepare the appropriate `Command` object. It is also reponsible for
+handling invalid inputs by the user. The `Parser` consist of the following methods.
+
+- `parse()`: Extracts out the command keyword from the user input.
+- `prepareCommand()`: Prepares the appropriate `Command` object based on the commmand keyword and the other
+relevant input fields provided by the user. It also checks if the format of the command is correct.
+- `isCorrectMarker()`: Checks if the marker used to label the input fields are correct.
+- `isValidImportance()`: Checks if the importance value provided by the user is a valid integer and within the [1, 10]
+range.
+- `isValidKeyword()`: Checks if the `keyWord` provided by the user is valid when preparing the `FilterCommand`.
+- `isValidCommand()`: Checks if the command keyword provided by the user is valid.
+
+Given below is an example of how the `Parser` works when it is issued a remove command.
+
+> **Step 1**: The user enters a command. The full command is read in by the `Ui`. `CLIAlgo` invokes the `parse()`
+> method from the `Parser`.
+
+> **Step 2**: The `parse()` method extracts out the command keyword provided by the user. It then calls the 
+> `prepareCommand()` method.
+
+> **Step 3**: The `parepareCommand()` identifies the correct `Command` object to prepare based on the command keyword.
+> Since the command keyword provided is `remove`, `prepareCommand()` calls `prepareRemoveCommand()`.
+
+> **Step 4**: If the `NAME` field of the command is null or not labelled using the correct marker, 
+> `prepareRemoveCommand()` returns an `InvalidCommand` object. If the name of the `CS2040CFile` provided is not
+> found, it returns a `NameNotFoundCommand`. If the `NAME` input field is valid, a `RemoveCommand` object is returned.
+
+The following sequence diagram shows how the Parser work.
+
+![](.\\sequence\\diagrams\\Parser.png "Parser Sequence Diagram")
```

Comment:
The step-by-step description of the Parser is easy to understand! Perhaps the sequence diagram can also be simplified such that it is easier to comprehend together with the description (e.g. by removing return arrows, which is easy to infer)? This could also be applied to the sequence diagrams for filtering by topic operation.